Boa.Constrictor/Screenplay/Waiting/AbstractWait.cs

```diff
@@ -143,19 +126,28 @@ protected TAnswer WaitForValue(IActor actor)
 
             // Verify successful waiting
             if (!satisfied)
-                throw new WaitingException<TAnswer>(this, actual);
-
-            // Return the actual awaited value
-            return actual;
+                ThrowWaitException();
```

Comment:
Thought: Could make WaitingException generic as it was before, with non-generic base?

Reply:
Just removed generic type from Wait as it was unnecessary. This makes this idea less likely.